x/protorev/keeper/routes_test.go

```diff
@@ -225,6 +246,99 @@ func (s *KeeperTestSuite) TestBuildHighestLiquidityRoute() {
 	}
 }
 
+// TestBuildTwoPoolRoute tests the BuildTwoPoolRoute function
+func (s *KeeperTestSuite) TestBuildTwoPoolRoute() {
+	cases := []struct {
+		description   string
+		swapDenom     types.BaseDenom
+		tokenIn       string
+		tokenOut      string
+		poolId        uint64
+		expectedRoute []TestRoute
+		hasRoute      bool
+	}{
+		{
+			description: "two pool route can be created with base as token out",
+			swapDenom: types.BaseDenom{
+				Denom:    types.OsmosisDenomination,
+				StepSize: sdk.NewInt(1_000_000),
+			},
+			tokenIn:  "stake",
+			tokenOut: types.OsmosisDenomination,
+			poolId:   53,
+			expectedRoute: []TestRoute{
+				{PoolId: 53, InputDenom: types.OsmosisDenomination, OutputDenom: "stake"},
+				{PoolId: 54, InputDenom: "stake", OutputDenom: types.OsmosisDenomination},
+			},
+			hasRoute: true,
+		},
+		{
+			description: "two pool route can be created with base as token in",
+			swapDenom: types.BaseDenom{
+				Denom:    types.OsmosisDenomination,
+				StepSize: sdk.NewInt(1_000_000),
+			},
+			tokenIn:  types.OsmosisDenomination,
+			tokenOut: "stake",
+			poolId:   53,
+			expectedRoute: []TestRoute{
+				{PoolId: 54, InputDenom: types.OsmosisDenomination, OutputDenom: "stake"},
+				{PoolId: 53, InputDenom: "stake", OutputDenom: types.OsmosisDenomination},
+			},
+			hasRoute: true,
+		},
+		{
+			description: "two pool route where swap is on the highest liquidity pool",
```

Comment:
i'm trying to understand this: if SwapDenom == tokenOutDenom & huge swap occours in highest liquidity pool, isn't based denom being cheaper so you **would** create a route to balance it?

Reply:
just to make sure i understand the scenario: 

if `BaseDenom` == `tokenOutDenom`, base denom becomes more expensive as the token supply of `baseDenom` in that pool decreases. i.e. if x == tokenInDenom and y == tokenOutDenom, then x + deltaX / y - deltaY > x / y. in this case, tokenOut is y which means it is more expensive and hence we can get more of X on the current pool than the other pool. We should then find the highest liquidity pool with the same assets and route through that. if the swap occurred on the highest liquidity pool, then we cannot make a route.